.github/workflows/veracode-scans.yml

```diff
@@ -0,0 +1,126 @@
+# Veracode pipeline scans
+on:
+  workflow_call:
+    inputs:
+      module_directory:
+        required: true
+        type: string
+      module_name:
+        required: true
+        type: string
+      major_version:
+        required: true
+        type: string
+      minor_version:
+        required: true
+        type: string
+      target_branch_name:
+        required: true
+        type: string
+      event_name:
+        required: true
+        type: string
+      threshold_pipeline:
+        required: true
+        type: string
+      threshold_sandbox:
+        required: true
+        type: string
+      cache_key:
+        required: true
+        type: string
+    secrets:
+      veracode_api_id:
+        required: true
+      veracode_api_key:
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    outputs:
+      scan_type: ${{ steps.routing-mode.outputs.scan_type }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get current time
+        run: echo "$(date +'%Y-%m-%d %H:%M:%S')"
+
+      - name: Generate binary file
+        uses: ./.github/actions/veracode-generate-binary
+        with:
+          module_directory: "${{ inputs.module_directory }}"
+          module_name: "${{ inputs.module_name }}"
+          target_branch_name: "${{ inputs.target_branch_name }}"
+          cache_key: '${{ inputs.cache_key }}'
+
+      - name: Set routing mode
+        id: routing-mode
+        if: inputs.event_name != 'pull_request' && (inputs.target_branch_name == 'develop' || inputs.target_branch_name == 'dev-[2-9][0-9].[0-9][0-9].x' || inputs.target_branch_name == 'master' || inputs.target_branch_name == '[2-9][0-9].[0-9][0-9].x')
```

Comment:
`inputs.target_branch_name == 'dev-[2-9][0-9].[0-9][0-9].x'` will not match `dev-23.04.x`
you should move the `if` expression in the bash, as it is done here : https://github.com/centreon/centreon/blob/develop/.github/workflows/get-version.yml#L90-L111